dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java

```diff
@@ -53,27 +60,89 @@
+        // Gather up any aliases that contain invalid prefixes
+        List<String> invalidAliases = aliases.stream().filter(alias -> StringUtils.startsWithAny(alias, INVALID_PREFIXES))
+                .collect(Collectors.toList());
 
-        // reserve some prefixes for our own use
-        String[] invalidPrefixes = {"dockstore", "doi", "drs", "trs", "dos", "wes"};
+        // If there are any aliases with invalid prefixes then report it to the user
+        if (invalidAliases.size() > 0) {
+            String invalidAliasesString = String.join(", ", invalidAliases);
+            String invalidPrefixesString = String.join(", ", INVALID_PREFIXES);
+            throw new CustomWebApplicationException("These aliases: " + invalidAliasesString + " start with a reserved string."
+                    + " They cannot be used. Please create aliases without these prefixes: " + invalidPrefixesString,
+                    HttpStatus.SC_BAD_REQUEST);
+        }
 
-        if (!user.isCurator() && !user.getIsAdmin()) {
-            for (String newAlias : newAliases) {
-                if (StringUtils.startsWithAny(newAlias, invalidPrefixes)) {
-                    throw new CustomWebApplicationException(newAlias + " starts with a reserved string, please try another alias",
+        if (blockAliasesWithZenodoFormat) {
+            List<String> aliasesWithForbiddenFormat = aliases.stream().filter(alias -> ZENODO_DOI_PATTERN.matcher(alias).matches())
+                    .collect(Collectors.toList());
+            // If there are any aliases with invalid formats then report it to the user
+            if (aliasesWithForbiddenFormat.size() > 0) {
+                String invalidAliasesString = String.join(", ", aliasesWithForbiddenFormat);
+                throw new CustomWebApplicationException("These aliases : " + invalidAliasesString + " have a format that is forbidden."
+                        + " They cannot be used. Please create aliases without this format: " + ZENDO_DOI_REGEX,
                         HttpStatus.SC_BAD_REQUEST);
-                }
             }
         }
+    }
+
+    /**
+     * Add aliases to an Entry (e.g. Workflow or Tool)
+     * and check that they are valid before adding them
+     * @param user user authenticated to issue a DOI for the workflow
+     * @param id the id of the Entry
+     * @param aliases a comma separated string of aliases
+     * @return the alias as a string
+     */
+    default T addAliases(User user, Long id, String aliases) {
+        return addAliasesAndCheck(user, id, aliases, true);
+    }
+
+    /**
+     * Add aliases to an Entry (e.g. Workflow or Tool)
+     * and check that they are valid before adding them:
+     * 1. If admin/curator, then no limit on prefix
+     * 2. If blockFormat false, then no limit on format
+     * @param user user authenticated to issue a DOI for the workflow
+     * @param id the id of the Entry
+     * @param aliases a comma separated string of aliases
```

Comment:
blockFormat is missing from JavaDoc. I feel like I mentioned this before, but maybe that was at a different spot, because I don't see the conversation here.

Reply:
Intellij paints the blockFormat JavaDoc string red...

Reply:
I think because I pass 'true' as the variable?

Reply:
fixed